### DISCUSSION TITLE
Disable updates for elements that are not visible?

### DISCUSSION THREAD

**Discussion by jeffective** [zauberzeug/nicegui#3744] [Q&A] [answered]
### Question

I am debugging some performance issues, and I find that I am updating a lot of elements that are not visible to the user (most of my elements are separated between three ui.tabs). How can I prevent them from being updated by bindings?

Perhaps I should disable ui.timer based on currently selected tab?

**Comment by python-and-novella** [comment]
for tabs, you can use `ui.tab_panels(keep_alive=False)` , it will make tab_panel destroied when it is not selected.

**Reply by jeffective** [reply]
I have found this is not the case, I am still getting websocket messages related to updates for elements in unselected ui.tabs (observed using network developer tools in chrome)

**Reply by falkoschindler** [reply]
`keep_alive` is a Vue feature and can reduce memory usage on the client. But the websocket updates are still sent.

**Comment by rodja** [comment, marked as answer]
We currently sync the full state to the client. Even the hidden elements. Mainly because it is simpler to implement and tracking / resyncing state when it gets visible may lead to very unexpected errors. I suggest you solve this in your application logic. For example by disabling timers.
